openmdao/drivers/doe_generators.py

```diff
@@ -0,0 +1,412 @@
+    _num_samples : int
+        The number of samples in the DOE.
+    """
+
+    _supported_criterion = [
+        "center", "c",
+        "maximin", "m",
+        "centermaximin", "cm",
+        "correlation", "corr",
+        None
+    ]
+
+    def __init__(self, samples=None, criterion=None, iterations=5, seed=None):
+        """
+        Initialize the LatinHypercubeGenerator.
+
+        See: https://pythonhosted.org/pyDOE/randomized.html
+
+        Parameters
+        ----------
+        samples : int, optional
+            The number of samples to generate for each factor (Defaults to n)
+        criterion : str, optional
+            Allowable values are "center" or "c", "maximin" or "m",
+            "centermaximin" or "cm", and "correlation" or "corr". If no value
+            given, the design is simply randomized.
+        iterations : int, optional
+            The number of iterations in the maximin and correlations algorithms
+            (Defaults to 5).
+        seed : int, optional
+            Random seed to use if design is randomized. Defaults to None.
+        """
+        super(LatinHypercubeGenerator, self).__init__()
+
+        if criterion not in self._supported_criterion:
+            raise ValueError("Invalid criterion '%s' specified for %s. "
+                             "Must be one of %s." %
+                             (criterion, self.__class__.__name__,
+                              self._supported_criterion))
+
+        self._samples = samples
+        self._criterion = criterion
+        self._iterations = iterations
+        self._seed = seed
+
+    def __call__(self, design_vars):
+        """
+        Generate case.
+
+        Parameters
+        ----------
+        design_vars : dict
+            Dictionary of design variables for which to generate values.
+
+        Yields
+        ------
+        list
+            list of name, value tuples for the design variables.
+        """
+        if self._seed is not None:
+            np.random.seed(self._seed)
+
+        size = sum([meta['size'] for name, meta in iteritems(design_vars)])
+
+        if self._samples is None:
+            self._samples = size
+
+        # generate design
+        doe = pyDOE2.lhs(size, samples=self._samples,
+                         criterion=self._criterion,
+                         iterations=self._iterations,
+                         random_state=self._seed)
+
+        self._num_samples = len(doe)
+
+        # generate values for each level for each design variable
+        # over the range of that varable's lower to upper bound
+        # rows = vars (# rows/var = var size), cols = levels
+        values = np.zeros((size, self._samples))
+
+        # yield desvar values for doe samples
+        for row in doe:
+            retval = []
+            col = 0
+            var = 0
+            for name, meta in iteritems(design_vars):
+                size = meta['size']
+                val = np.zeros(size)
```

Comment:
could use np.empty here instead of np.zeros since you set every entry of it later

Reply:
done